src/main/java/io/kestra/plugin/notifications/services/ExecutionService.java

```diff
@@ -13,14 +13,16 @@
 import io.kestra.core.utils.RetryUtils;
 import io.kestra.core.utils.UriProvider;
 import io.kestra.plugin.notifications.ExecutionInterface;
+import jakarta.annotation.Nullable;
 
 import java.time.Duration;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.NoSuchElementException;
+import java.util.Optional;
 
 public class ExecutionService {
-    public static Execution findExecution(RunContext runContext, Property<String> executionId, Property<String> executionStateWhenTriggerWasTriggered) throws IllegalVariableEvaluationException, NoSuchElementException {
+    public static Execution findExecution(RunContext runContext, Property<String> executionId, @Nullable Property<String> executionStateWhenTriggerWasTriggered) throws IllegalVariableEvaluationException, NoSuchElementException {
```

Comment:
```suggestion
    public static Execution findExecution(RunContext runContext, Property<String> executionId, @Nullable Property<String> triggerExecutionState) throws IllegalVariableEvaluationException, NoSuchElementException {
```

Just a suggestion, but the name is pretty big